### PR TITLE
fix(gemini): apply model execution config

### PR DESCRIPTION
**Pull request by Zbhbb**
## AgentScope-Java Version

Current main branch.

### Background

`GeminiChatModel` returned its SDK stream directly, so configured request timeout and retry settings were ignored even though other providers apply the shared execution wrapper.

### Changes

- Apply `ModelUtils.applyTimeoutAndRetry` to Gemini response streams.
- Add timeout and retry coverage through the public `stream()` entry point.

### Validation

- `mvn -q -pl agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini -am -Dtest=GeminiChatModelTest test`
- `mvn -q -T1 -pl agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini -am verify`

The full repository `verify` ran through its Java test phase, but the unrelated `agentscope-builder` step could not download npm 10.9.2 because the npm registry TLS handshake was terminated.

## Checklist

- [x] Code has been formatted with `mvn spotless:apply` rules
- [x] Relevant tests are passing
- [x] Javadoc follows existing conventions
- [ ] Documentation update is not needed
- [x] Code is ready for review

### CODE REVIEW

File: agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java [hunk 1]
```diff
@@ -58,6 +66,58 @@ void setUp() {
...
+                                        .maxAttempts(1)
+                                        .build())
+                        .build();
+
+        ModelException error =
+                assertThrows(
+                        ModelException.class,
+                        () ->
+                                new TestGeminiChatModel(Flux.never())
+                                        .stream(List.of(), List.of(), options).blockLast());
```
[the remark this answers is not in the document]
Comment (Zbhbb) [reply agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java]: Fixed in 062372bd. The test now uses blockLast(Duration.ofSeconds(1)), so a missing execution wrapper fails within a bounded interval while the expected ModelException still proves Gemini's configured timeout is applied. Verified with the Gemini focused test and module verify.

File: agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java [hunk 2]
```diff
@@ -629,4 +689,22 @@ private static HttpOptions getHttpOptions(GeminiChatModel model) throws Exceptio
...
+        private TestGeminiChatModel(Flux<ChatResponse> response) {
+            super("test-key", "gemini-test", true, null, null, null, null, null, null, null, null);
+            this.response = response;
+        }
+
+        @Override
+        protected Flux<ChatResponse> doStream0(
+                List<io.agentscope.core.message.Msg> messages,
+                List<ToolSchema> tools,
+                GenerateOptions options) {
```
[the remark this answers is not in the document]
Comment (Zbhbb) [reply agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java]: Fixed in 062372bd. Added the Msg import and changed the override signature to List<Msg>.

**Review by oss-maintainer — APPROVED** [review]
LGTM. Looks good.

---
*Automated review by "github-manager-bot"*

File: agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java [hunk 3]
```diff
@@ -629,4 +691,20 @@ private static HttpOptions getHttpOptions(GeminiChatModel model) throws Exceptio
         httpOptionsField.setAccessible(true);
         return (HttpOptions) httpOptionsField.get(model);
     }
+
+    private static final class TestGeminiChatModel extends GeminiChatModel {
+
+        private final Flux<ChatResponse> response;
+
+        private TestGeminiChatModel(Flux<ChatResponse> response) {
+            super("test-key", "gemini-test", true, null, null, null, null, null, null, null, null);
```
Comment (AgentScopeJavaBot) [inline agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java:700]: [major] TestGeminiChatModel super() passes 11 args but GeminiChatModel constructor expects 12 (baseUrl param added on main after branch). Will not compile after rebase. Add null as 2nd arg: `super("test-key", null, "gemini-test", true, null, null, null, null, null, null, null, null);`

File: agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/main/java/io/agentscope/extensions/model/gemini/GeminiChatModel.java [hunk 4]
```diff
@@ -232,6 +233,17 @@ private static HttpOptions resolveHttpOptions(String baseUrl, HttpOptions httpOp
     @Override
     protected Flux<ChatResponse> doStream(
             List<Msg> messages, List<ToolSchema> tools, GenerateOptions options) {
+        return applyExecutionConfig(doStream0(messages, tools, options), options);
+    }
+
+    private Flux<ChatResponse> applyExecutionConfig(
```
Comment (AgentScopeJavaBot) [inline agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/main/java/io/agentscope/extensions/model/gemini/GeminiChatModel.java]: [nitpick] Private `applyExecutionConfig` wrapper is unnecessary indirection; all other models call `ModelUtils.applyTimeoutAndRetry` directly in doStream. Consider inlining for consistency.

**Review by AgentScopeJavaBot — COMMENTED** [review]
🤖 AI Review

A focused fix that brings `GeminiChatModel` in line with the established timeout/retry pattern used by all other model implementations. The `doStream` → `doStream0` extraction is clean and the tests are well-designed. One issue requires attention: the `TestGeminiChatModel` super-constructor call has 11 args but current constructor expects 12 (due to `baseUrl` being added on main after this branch was created).

File: agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java (hunk 3 above)
Comment (AgentScopeJavaBot) [inline agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java:700]: [major] TestGeminiChatModel super() passes 11 args but GeminiChatModel constructor expects 12 (baseUrl param added on main after branch). Will not compile after rebase. Add null as 2nd arg: `super("test-key", null, "gemini-test", true, null, null, null, null, null, null, null, null);`

File: agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/main/java/io/agentscope/extensions/model/gemini/GeminiChatModel.java (hunk 4 above)
Comment (AgentScopeJavaBot) [inline agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/main/java/io/agentscope/extensions/model/gemini/GeminiChatModel.java]: [nitpick] Private `applyExecutionConfig` wrapper is unnecessary indirection; all other models call `ModelUtils.applyTimeoutAndRetry` directly in doStream. Consider inlining for consistency.

**Review by AgentScopeJavaBot — COMMENTED** [review]
🤖 AI Review

A focused fix that brings `GeminiChatModel` in line with the established timeout/retry pattern used by all other model implementations. The `doStream` → `doStream0` extraction is clean and the tests are well-designed. One issue requires attention: the `TestGeminiChatModel` super-constructor call has 11 args but current constructor expects 12 (due to `baseUrl` being added on main after this branch was created).

File: agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java (hunk 3 above)
Comment (AgentScopeJavaBot) [inline agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java:700]: [major] TestGeminiChatModel super() passes 11 args but GeminiChatModel constructor expects 12 (baseUrl param added on main after branch). Will not compile after rebase. Add null as 2nd arg: `super("test-key", null, "gemini-test", true, null, null, null, null, null, null, null, null);`

File: agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/main/java/io/agentscope/extensions/model/gemini/GeminiChatModel.java (hunk 4 above)
Comment (AgentScopeJavaBot) [inline agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/main/java/io/agentscope/extensions/model/gemini/GeminiChatModel.java]: [nitpick] Private `applyExecutionConfig` wrapper is unnecessary indirection; all other models call `ModelUtils.applyTimeoutAndRetry` directly in doStream. Consider inlining for consistency.

**Review by AgentScopeJavaBot — COMMENTED** [review]
🤖 AI Review

A focused fix that brings `GeminiChatModel` in line with the established timeout/retry pattern used by all other model implementations. The `doStream` → `doStream0` extraction is clean and the tests are well-designed. One issue requires attention: the `TestGeminiChatModel` super-constructor call has 11 args but current constructor expects 12 (due to `baseUrl` being added on main after this branch was created).

File: agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java (hunk 3 above)
Comment (Zbhbb) [reply agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java:700]: Verified against the current main: GeminiChatModel retains the backward-compatible 11-argument constructor, which delegates to the 12-argument constructor with baseUrl = null. After rebasing this PR onto current main, the focused GeminiChatModelTest command passes. No constructor change is required.

Comment (Zbhbb) [reply agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java:700]: Verified on current main: GeminiChatModel retains an 11-argument constructor that delegates to the 12-argument form with baseUrl = null. The test compiles after rebase and the focused GeminiChatModelTest passes, so adding a redundant null is unnecessary.

File: agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/main/java/io/agentscope/extensions/model/gemini/GeminiChatModel.java (hunk 4 above)
Comment (Zbhbb) [reply agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/main/java/io/agentscope/extensions/model/gemini/GeminiChatModel.java]: applyExecutionConfig is intentionally kept as the public doStream boundary for the shared timeout and retry policy. doStream0 remains the provider and test seam; inlining would not alter behavior and would make that boundary less explicit. No functional change is needed.

File: agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java (hunk 3 above)
Comment (AgentScopeJavaBot) [reply agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java:700]: ✅ Verified as addressed in `1b8cebd`: Bot's premise is wrong: GeminiChatModel has an 11-parameter constructor (line 182, no baseUrl) that matches the test super() call exactly; code compiles fine.

Comment (AgentScopeJavaBot) [reply agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java:700]: ✅ Verified as addressed in `1b8cebd`: 11-parameter constructor exists and matches the test super() call; bot's 12-arg concern is invalid.

Comment (AgentScopeJavaBot) [reply agentscope-extensions/agentscope-extensions-model/agentscope-extensions-model-gemini/src/test/java/io/agentscope/extensions/model/gemini/GeminiChatModelTest.java:700]: ✅ Verified as addressed in `1b8cebd`: 11-parameter constructor confirmed at line 182; test super() call with 11 args matches it correctly.

**Review by AgentScopeJavaBot — APPROVED** [review]
All blocking review comments have been addressed. 3 non-blocking suggestion(s) (minor/nit) remain open and may be addressed in a follow-up.
